Source/CarthageKit/Xcode.swift

```diff
@@ -680,33 +679,43 @@ public func buildInDirectory(_ directoryURL: URL, withOptions options: BuildOpti
 					} else {
 						filteredSDKs = sdks.filter { options.platforms.contains($0.platform) }
 					}
-
 					return sdkFilter(filteredSDKs, scheme, configuration, project)
 				}
 
-				let buildProgress = buildScheme(scheme, withConfiguration: options.configuration, inProject: project, workingDirectoryURL: directoryURL, derivedDataPath: options.derivedDataPath, toolchain: options.toolchain, sdkFilter: wrappedSDKFilter)
+				return buildScheme(scheme, withOptions: options, inProject: project, workingDirectoryURL: directoryURL, sdkFilter: wrappedSDKFilter)
 					.mapError { (error) -> CarthageError in
 						if case let .taskError(taskError) = error {
 							return .buildFailed(taskError, log: nil)
 						} else {
 							return error
 						}
 					}
-					// Discard any existing Success values, since we want to
-					// use our initial value instead of waiting for
-					// completion.
-					.map { taskEvent in
-						return taskEvent.map { _ in initialValue }
-					}
-					.filter { taskEvent in taskEvent.value == nil }
-
-				return BuildSchemeProducer(value: .success(initialValue))
-					.concat(buildProgress)
+					.on(started: {
+						observer.send(value: .success(initialValue))
+					})
```

Comment:
this runs into the old "Need to collect the task events but also want to send them through" issue (the stdout/stderr portion was resolved by `collectTaskEvents`)

So here, the original code basically inverted the 'usual' task output. It forced the 'initialValue' to be sent at the beginning of the task instead of the end, and then used map/filter to throw away the 'real' task event. In the current codebase it was ok to throw away the real task value, it was never actually used. The problem is that real task value is the `URL` values we want to collect on. 

So if I tack it onto the beginning, it'll get caught up in the collect (since it's not a stdout event). If I map to it after the collect, it'll wait til the build is done. I think the real solution would be to get rid of the 'invert the task event' workaround altogether, and maybe somehow send out an event to the project events observer, or something higher up. But as it is now, there's no real way to "tell" anyone the build started while still waiting for the build to finish in the same producer.